evennia/contrib/aisystem/README.md

```diff
@@ -0,0 +1,661 @@
+
+@aiassigntree - assigns the specified BehaviorTree as the origin tree of a given
+                agent.
+
+@ailist - shows a list of all BehaviorTrees currently in the game.
+
+@ailook - displays a node's data, a node instance's data or the global data of
+          a given agent.
+
+@aistatus - displays what tree, node and agent the player is currently browsing
+
+@aiwatch - adds a node instance to the player's watchlist, ensuring that
+           information about it is displayed whenever the node is run.
+
+@aiunwatch - removes a node instance from the player's watchlist
+
+@aigo - sets the player's browsing cursor to a given tree and node
+
+@aibb - sets the player's browsing cursor to a given agent
+
+@aiup - moves the browsing cursor to the currently browsed node's parent
+
+@aidown - moves the browsing cursor to one of the currently browsed node's
+          children
+
+@aisetprop - assigns a value to a node's property, a node instance's entry or
+             an agent's global data
+
+@aidelprop - removes a node's property, a node instance's entry or an agent's
+             global data
+
+@aiaddnode - adds a node of the specified type to a location in the target tree
+
+@aicopynode - attaches a copy of the specified node to another node
+
+@aimovenode - moves a specified node under another node
+
+@aiswap - swaps the locations of two nodes
+
+@aishift - shifts the position of a node in its parent node's list of children
+
+@airemove - removes a node from its tree
+
+Note that, at present, there is no way to convert AI trees stored in the
+game's database into Python code. I can make that feature available on
+request, however.
+
+### Adding your own node types
+
+If you are not a coder, this section will not be useful to you.
+
+The AI system provides barebones functionality, but you will want to create
+a set of nodes tailored to your own game. To this end, you need to subclass
+either one of the fully functional node classes that come with the AI system,
+or the LeafNode, DecoratorNode and CompositeNode classes, which have no
+functionality. You should never subclass your nodes from the Node or RootNode
+class.
+
+Each node you create has a `weight` property that should not be overriden, but
+can be assigned a float value to determine how likely this node is to be picked
+by a ProbSelector or ProbSequence node.
+
+Each node also has a `name` property that contains the name you assigned it,
+as well as a `hash` property, which contains three alphanumeric characters that
+are unique to the node within a given tree, followed by an underscore character
+and the id number of its tree. For instance: `H5q_1566`, `Zmr_43`, `7Fl_5425163`
+The first three characters of this hash are used by the AI system's various
+commands and can be used to reference the nodes when two or more nodes in the
+tree share the same name.
+
+Each node contains the following methods, all of which may be overridden.
```

Comment:
This could well be a new subsection, for example "Properties of the Node object".  All the methods listing within should probably go in a list for better readability in the formatted version.